src/shared/Utilities/ByteBuffer.h

```diff
@@ -430,6 +430,25 @@ class ByteBuffer
             return *this;
         }
 
+		/**
+		 * @brief
+		 *
+		 * @param value
+		 * @return ByteBuffer &operator >>
+		 */
+		ByteBuffer& operator>>(char *pval)
+		{
+			while (rpos() < size())                         // prevent crash at wrong string format in packet
+			{
+				char c = read<char>();
+
+				*pval++ = c;
```

Comment:
Pay attention to buffer overflow. Why do we need this if we already have operator >> (std::string&)? Give a short scenario for it's usage.

Reply:
In the spirit of buffer overflow, recall we should never trust the client packets. What happens if a malformed packet, with no null char, is sent? What happens with pval array after the while loop in that case?

Reply:
Yep, my fix corrects both problems. Thanks for the heads up. <3